src/systems/equation_systems.C

```diff
@@ -855,65 +856,104 @@ EquationSystems::build_elemental_solution_vector (std::vector<Number> & soln,
 
   // Localize into 'soln', provided that parallel_soln is not empty.
   // Note: parallel_soln will be empty in the event that none of the
-  // input names were CONSTANT, MONOMIAL variables or there were
-  // simply no CONSTANT, MONOMIAL variables in the EquationSystems
-  // object.
+  // input names were CONSTANT, MONOMIAL nor components of CONSTANT,
+  // MONOMIAL_VEC variables or there were simply none of these in
+  // the EquationSystems object.
   soln.clear();
   if (parallel_soln)
     parallel_soln->localize_to_one(soln);
 }
 
-
-
 std::vector<std::pair<unsigned int, unsigned int>>
 EquationSystems::find_variable_numbers
-  (std::vector<std::string> & names, const FEType * type) const
+  (std::vector<std::string> & names, const FEType * type, const std::vector<FEType> * types) const
 {
   // This function must be run on all processors at once
   parallel_object_only();
 
   libmesh_assert (this->n_systems());
 
-  // If the names vector has entries, we will only populate the soln vector
-  // with names included in that list.  Note: The names vector and var_nums
-  // vector pairs are reordered alphabetically upon exiting this function.
-  std::vector<std::pair<unsigned int, unsigned int>> var_nums;
-  std::vector<std::string> filter_names = names;
-  bool is_filter_names = !filter_names.empty();
-
+  // Resolve class of type input and ensure that at least one of them is null
+  libmesh_error_msg_if(type && types,
+                       "Input either 'type' or 'types' in find_variable_numbers, but not both.");
```

Comment:
Should this be libmesh_assert_msg(), since although we fear bad code is going to try to pass in two non-nullptr arguments here, there should be no way for a bad input file to do so?

Reply:
Yeah it's the developer's fault if this happens. So okay, assert makes more sense.